tests/system/small/test_index.py

```diff
@@ -320,6 +323,33 @@ def test_index_to_series(
     pd.testing.assert_series_equal(bf_result, pd_result)
 
 
+def test_index_to_frame():
+    index_to_frame_index_args = [True, False]
+    index_to_frame_name_args = [None, "food"]
+    pd_idx: pd.Index = pd.Index(
+        ["Ant", "Bear", "Cow"], name="animal", dtype="string[pyarrow]"
+    )
+    bf_idx = indexes.Index(["Ant", "Bear", "Cow"], name="animal")
+
+    for index_arg, name_arg in itertools.product(
```

Comment:
you can use @pytest.mark.parametrize to achieve combinations of params.

Which creates multiple separate tests. Easier to find the problem if anyone fails.